docs/Coding_Conventions_and_Standards.md

```diff
@@ -12,29 +12,93 @@ Google style from <https://google.github.io/styleguide/cppguide.html> with a few
   * Allowed
   * Use a non-const reference for arguments that are modifiable but cannot be `nullptr` so the API clearly advertises the intent
   * Const correctness and usage of smart pointers (`shared_ptr` and `unique_ptr`) is expected. A non-const reference equates to "this is a non-null object that you can change but are not being given ownership of".
+* Prefer passing `gsl::span<const T>` by value (or `std::span` when supported) as input arguments when passing const references to containers with contiguous storage (like `std::vector`). This allows the function to be container independent, and the argument to represent arbitrary memory spans or sub-spans. The below examples allow the client code to use either `std::vector` or `InlinedVector`. An instance of a `gsl::span` would be created automatically.
+```cpp
+/// Instead of
+void foo(const std::vector<int64_t>&);
+
+/// Use to pass any contiguous const container containing int64_t
+// Now you can seamlessly pass either `std::vector`, `InlinedVector`, `std::array` or `gsl::span` as an argument.
+void foo(gsl::span<const int64_t>);
+
+// Example with pointer to const data. Instead of
+void foo(const std::vector<const Node*>&);
+
+// Use
+void foo(gsl::span<const Node* const>);
+```
+* Prefer returning `gsl::span<const T>` by value instead of a const reference to a contiguous member container. Prefer returning `gsl::span` instead of a pointer referring to a chunk of memory. The size is also included in the span.
```

Comment:
```suggestion
* Prefer returning `gsl::span<const T>` by value instead of a const reference to a contiguous member container. Prefer returning `gsl::span` instead of a pointer referring to a chunk of memory as the size is also included in the span.
```
as suggested by @skottmckay